services/workflows-service/src/storage/get-file-storage-manager.ts

```diff
@@ -0,0 +1,71 @@
+import { diskStorage } from 'multer';
+import { getFileName } from '@/storage/get-file-name';
+import { z } from 'zod';
+import multerS3 from 'multer-s3';
+import { S3Client, S3ClientConfig, GetObjectCommand } from '@aws-sdk/client-s3';
+import * as tmp from 'tmp';
+import * as fs from 'fs';
+
+import { Readable } from 'stream';
+export type TLocalFile = string;
+
+const isS3BucketConfigured = () => {
+  return process.env.AWS_S3_BUCKET_KEY;
+};
+
+const generateAwsConfig = (): S3ClientConfig => {
+  return {
+    region: z.string().parse(process.env.AWS_REGION),
+    credentials: {
+      accessKeyId: z.string().parse(process.env.AWS_S3_BUCKET_KEY),
+      secretAccessKey: z.string().parse(process.env.AWS_S3_BUCKET_SECRET),
+    },
+  };
+};
+
+export const fetchDefaultBucketName = () => {
```

Comment:
I think we could access `process.env.AWS_S3_BUCKET_NAME` here without a function, especially if it was already validated. Otherwise changing `fetch` to `get` would be clearer to me.